bats/scripts/rename-logs.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+# GitHub restricts artifact filenames:
+
+# Invalid characters include: Double quote ", Colon :, Less than <,
+# Greater than >, Vertical bar |, Asterisk *, Question mark ?, Carriage
+# return \r, Line feed \n
+#
+# The following characters are not allowed in files that are uploaded
+# due to limitations with certain file systems such as NTFS. To maintain
+# file system agnostic behavior, these characters are intentionally not
+# allowed to prevent potential problems with downloads on different file
+# systems.
+
+if [[ -z ${1:-} ]]; then
+    echo "usage: $0 LOGDIR"
+    exit 1
+fi
+
+# Find all files and put the names into the FILES array.
+# We don't rename inside the loop to make sure the find command has
+# finished before we modify any directories it is iterating over.
+FILES=()
+while read -d $'\0' -r FILE; do
+    FILES+=("$FILE")
+done < <(find "$1" -type f -print0)
+
+# URL-Encode all the forbidden characters
+# (bats has already URL-encoded the slash)
+for FILE in "${FILES[@]}"; do
```

Comment:
Just realized that this will break down horribly if the _directory_ (or rather any point up to it) contains a banned character. That's probably okay though; if it fails it will do so loudly.

Reply:
Yeah, let's get this merged now. It is not a general filename encoding utility, but a hack to make sure our `@test` description, when used as filenames by `--gather-test-outputs-in`, can be uploaded by Github actions.